ibm/data_source_ibm_cis_healthchecks.go

```diff
@@ -0,0 +1,171 @@
+							Type:        schema.TypeString,
+							Description: "method",
+							Computed:    true,
+						},
+						cisGLBHealthCheckTimeout: {
+							Type:        schema.TypeInt,
+							Description: "timeout",
+							Computed:    true,
+						},
+						cisGLBHealthCheckRetries: {
+							Type:        schema.TypeInt,
+							Description: "retries",
+							Computed:    true,
+						},
+						cisGLBHealthCheckInterval: {
+							Type:        schema.TypeInt,
+							Description: "interval",
+							Computed:    true,
+						},
+						cisGLBHealthCheckFollowRedirects: {
+							Type:        schema.TypeBool,
+							Description: "follow_redirects",
+							Computed:    true,
+						},
+						cisGLBHealthCheckAllowInsecure: {
+							Type:        schema.TypeBool,
+							Description: "allow_insecure",
+							Computed:    true,
+						},
+						cisGLBHealthCheckCreatedOn: {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						cisGLBHealthCheckModifiedOn: {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						cisGLBHealthCheckPort: {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMCISGLBHealthCheckRead(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).CisGLBHealthCheckClientSession()
+	if err != nil {
+		return err
+	}
+
+	crn := d.Get(cisID).(string)
+	sess.Crn = core.StringPtr(crn)
+
+	opt := sess.NewListAllLoadBalancerMonitorsOptions()
+
+	result, resp, err := sess.ListAllLoadBalancerMonitors(opt)
+	if err != nil {
+		log.Printf("Error listing global load balancer health check detail: %s", resp)
+		return err
+	}
+
+	monitors := make([]map[string]interface{}, 0)
+	for _, instance := range result.Result {
+		monitor := map[string]interface{}{}
+		monitor["id"] = convertCisToTfTwoVar(*instance.ID, crn)
+		monitor[cisGLBHealthCheckID] = *instance.ID
+		monitor[cisGLBHealthCheckDesc] = *instance.Description
+		monitor[cisGLBHealthCheckPath] = *instance.Path
+		monitor[cisGLBHealthCheckCreatedOn] = *instance.CreatedOn
+		monitor[cisGLBHealthCheckModifiedOn] = *instance.ModifiedOn
+		monitor[cisGLBHealthCheckExpectedBody] = *instance.ExpectedBody
+		monitor[cisGLBHealthCheckExpectedCodes] = *instance.ExpectedCodes
+		monitor[cisGLBHealthCheckType] = *instance.Type
+		monitor[cisGLBHealthCheckMethod] = *instance.Method
+		monitor[cisGLBHealthCheckTimeout] = *instance.Timeout
+		monitor[cisGLBHealthCheckRetries] = *instance.Retries
+		monitor[cisGLBHealthCheckInterval] = *instance.Interval
+		monitor[cisGLBHealthCheckFollowRedirects] = *instance.FollowRedirects
+		monitor[cisGLBHealthCheckAllowInsecure] = *instance.AllowInsecure
+		monitor[cisGLBHealthCheckPort] = 0
+		if instance.Port != nil {
+			monitor[cisGLBHealthCheckPort] = *instance.Port
+		}
+
+		monitors = append(monitors, monitor)
+	}
+	d.SetId(dataSourceIBMCISGLBHealthCheckID(d))
+	d.Set(cisGLBHealthCheck, monitors)
```

Comment:
set required parameters as well
`crn`

Reply:
do I need to keep the ```crn``` since, it is an input to datasource?

Reply:
yes, please set all the attributes that are in schema..

Reply:
done